lib/avo/fields/base_field.rb

```diff
@@ -93,7 +93,11 @@ def initialize(id, **args, &block)
       end
 
       def translation_key
-        @translation_key || "avo.field_translations.#{@id}"
+        return @translation_key if @translation_key.present?
+
+        return "#{@resource.base_field_translation_key}.#{@id}" if @resource.try(:base_field_translation_key)
+
+        "avo.field_translations.#{@id}"
```

Comment:
In a palliative way and trying to avoid DRY I used it this way:

```yml
pt-BR:
  activerecord:
    models:
      user: &user_names
        zero: "Usuários"
        one: "Usuário"
        other: "Usuários"
    attributes:
      user: &user_fields
        name: "Nome"
        email: "E-mail"
        password: "Senha"
  avo:
    resource_translations:
      <<: *user_names
    field_translations:
      <<: *user_fields
```

however I think you should consider translation from model, maybe like this:


```suggestion
        return @translation_key if @translation_key.present?

        return "activerecord.attributes.#{@resource.record.model_name.singular}.#{@id}" if @resource.record.respond_to?(@id)

        "avo.field_translations.#{@id}"
```

If you accept this suggestion maybe you should remove `class_attribute :base_field_translation_key` at `lib/avo/base_resource.rb:51`

---

PS: I know it'll change the standard, but it'll follow the rails conversion and best of all, we wouldn't need to rewrite the translation of the models (For who are migrating from other projects)

Reply:
Hey guys, let me pitch in with an opinion.

I'm going to preface this by saying that I did not know about Rails' way of translating models (`en.active_record.models.user`) when I first wrote the i18n part for Avo.

But, after a while of spending time with Avo, I think it's better that we didn't use that as a resource is not always a one-to-one representative.
One can use on model with multiple resources, thus needing multiple translations for something that active record only has once.

I do see that if you already translated the `activerecord` model, you wouldn't want to translate it again.
But there is a solution for that. 
You could se the [`translation_key`](https://docs.avohq.io/3.0/localization.html#localizing-resources) on a resource with the translation key that you already have `self.translation_key = `activerecord.models.user`.
Avo will use that to translate the resource.

Reply:
Hello. When you say translate the resource, you are talking about the resource name. If you intend to translate fields, you need to repeat translation_key a lot. The issue here is that fields are in the root of fields_translations. This is very different from the community standards and Avo is here to simplify things and speed up development, right?

This is the pitch here. 😇

Reply:
Ok. I think I got it.

The pitch here is "give me a setting that will make Avo i18n behave like Rails' so I don't have to manually declare the translation keys for all my fields".

Am I correct with that statement?